tests/adapters/test_adapter.py

```diff
@@ -22,25 +22,52 @@
 # THE SOFTWARE.
 #
 
-import logging
+from unittest import mock
 
-from pymeasure.adapters import FakeAdapter
+import pytest
 
-log = logging.getLogger(__name__)
-log.addHandler(logging.NullHandler())
+from pymeasure.adapters import Adapter, FakeAdapter
 
 
-def test_adapter_values():
+@pytest.fixture()
+def adapter():
+    return Adapter(preprocess_reply=lambda v: v)
+
+
+def test_init(adapter):
+    assert adapter.preprocess_reply is not None
+    assert adapter.connection is None
+
+
+def test_del(adapter):
+    adapter.connection = mock.MagicMock()
+    adapter.__del__()
+    assert adapter.connection.method_calls == [mock.call.close()]
+
+
+@pytest.mark.parametrize("method, args", (['write', ['5']], ['read', []], ['binary_values', ['8']]))
+def test_not_implemented_methods(adapter, method, args):
+    with pytest.raises(NameError):
+        getattr(adapter, method)(*args)
+
+
+@pytest.mark.parametrize("value, kwargs, result",
+                         (("5,6,7", {}, [5, 6, 7]),
+                          ("5.6.7", {'separator': '.'}, [5, 6, 7]),
+                          ("5,6,7", {'cast': str}, ['5', '6', '7']),
+                          ("X,Y,Z", {}, ['X', 'Y', 'Z']),
+                          ("X,Y,Z", {'cast': str}, ['X', 'Y', 'Z']),
+                          ("X.Y.Z", {'separator': '.'}, ['X', 'Y', 'Z']),
+                          ("0,5,7.1", {'cast': bool}, [False, True, True]),
+                          ))
+def test_adapter_values(value, kwargs, result):
     a = FakeAdapter()
-    assert a.values("5,6,7") == [5, 6, 7]
-    assert a.values("5,6,7", cast=str) == ['5', '6', '7']
-    assert a.values("X,Y,Z") == ['X', 'Y', 'Z']
-    assert a.values("X,Y,Z", cast=str) == ['X', 'Y', 'Z']
-    assert a.values("X.Y.Z", separator='.') == ['X', 'Y', 'Z']
+    assert a.values(value, **kwargs) == result
 
 
 def test_adapter_preprocess_reply():
     a = FakeAdapter(preprocess_reply=lambda v: v[1:])
+    str(a)
```

Comment:
?

Reply:
Just verifying, that the representation is called (code coverage...), yeah, I should assert it.